preview_generator/preview/builder/image__imconvert.py

```diff
@@ -53,7 +53,7 @@ class ImagePreviewBuilderIMConvert(ImagePreviewBuilder):
         MimetypeMapping("image/heic", ".heif"),
     ]
 
-    weight = 30
+    weight = 0
```

Comment:
- I have checked that when I set the image size to 1300 * 1300 the colour differences look very subtle and hard to distinguish, (maybe my monitor is not rich enough to display the colours), also I have tried forcing colorspace=srgb or rgb and the colours are more distorted.
- yes, it has been added
- I have updated the supported_mimetypes.rst file, but I don't know why the previous rst file shows that the ImageMagick cmd builder can support so many files, even if I go back to the previous commit fdce64a84cf1b48892d534ff917e3e2d3910f89a I can't rebuild the list to support the same number of formats

Reply:
And I did  find that the format returned  by  this `get_supported_mimetypes`  function  of `ImagePreviewBuilderWand ` or ImagePreviewBuilderIMConvert was much less than [the format supported by ImageMagick itself](https://imagemagick.org/script/formats.php). Maybe we need refactor this function.

Reply:
Thanks for theses changes. There is still few issues, but i don't think we should solve all of these for now, just keep same feature with wand as previously and solve the other problem later:

i created a branch with the updated list on my system + some others fix: https://github.com/algoo/preview-generator/tree/feat/wand. Can you get the fix from it and update this PR ?

* for the color issue, i suggest creating a new ticket about it, and explain what you tried and give the result obtain.

* for the supported mimetype list file, it look like the list is really dependent on system and imagemagick version itself, so the list may diverge. i updated the list from my system on my branch, so you can merge it for now, and we may later do some check to understand with the list is different between system.

* for the fact that get_supported_mimetype is not accurate, there is maybe a bug somewhere. i created a ticket about it: https://github.com/algoo/preview-generator/issues/277 to limit the amount of work for this PR.

* You set the wand builder as 40 priority same as libreoffice, this is a bad practice as this doesn't permit to know which one of those two will be used if there is a conflict. I fixed this in my branch.

* About the raw support, i re-added the support on my branch:
  * a better builder that don't rely on the old will be needed in a not so far future (see https://github.com/algoo/preview-generator/issues/249)
  * i added a test to show that there is currently an issue with some raw format without extension in the file itself ( note: there is a bug about removing this feature here: https://github.com/algoo/preview-generator/issues/260), can you check if possible to force the guessing of type in wand ?
  *  for unclear the wand support of xcf is really poor (pillow give better result). I don't get it for now.
  * i did not check heic format support for now.

Reply:
About the xcf support, it's really strange:
- `convert -flatten What_a_Brilliant_Idea\!\ _Barnstar.xcf myfile.jpg` locally give me good result:
![myfile](https://user-images.githubusercontent.com/7851652/140074573-ff34f019-ee1f-48eb-8ca8-e1c21c87998d.jpg)
- `preview What_a_Brilliant_Idea\!\ _Barnstar.xcf` from preview generator give me full white result:
![fc6b1f63524bbb3a3e1bb4d7bd612b94-256x256](https://user-images.githubusercontent.com/7851652/140074952-27902cfc-8957-4c32-a908-5f4b38a18e6b.jpeg) 
Yes there is an image ^ here.

Reply:
>i added a test to show that there is currently an issue with some raw format without extension in the file itself ( note: there is a bug about removing this feature here: #260), can you check if possible to force the guessing of type in wand ?

Yeah, I have tried it, indeed, without the extension, IM really can’t guess the ARW file format well, [see here for details.](https://legacy.imagemagick.org/discourse-server/viewtopic.php?t=28551)
But WAND has a way to hint the file extension.  Refer: 
https://docs.wand-py.org/en/0.6.7/guide/security.html?highlight=extension#security
https://docs.wand-py.org/en/0.6.7/guide/read.html#hint-file-format 

But I ran into a problem. The `build_jpeg_preview` method currently does not effectively accept a user-specified `file_ext` parameter.
It has a similar parameter `extension`, but it doesn't seem to be the same thing at all. `extension` is to specify the format of the desired output file rather than the format of the input file
And the `extension` parameter seems to be useless (it is currently almost a fixed value)

I might do this:

Add a `file_ext` parameter to the function `build_jpeg_preview`, the value of this `file_ext` comes from the `file_ext` parameter of PreviewManager's `get_jpeg_preview`
At the same time, is the `extension` parameter temporarily retained or deleted?
What do you think?
There may be another option, `build_jpeg_preview` accepts both `input_ex`t and `out_ext`. That is, the original `extension` is renamed to `out_ex`t, and the `input_ext` parameter is added

Reply:
For now, i suggest to just follow the "hacky" way we do this in current imconvert builder.
We should probably refactor this part later as this is far from practical, but the current idea is that:
- we have file_ext at manager level
- we have mimetype at builder level based on the file_ext given at manager level
- we infer the file_ext from mimetype at builder level:

https://github.com/algoo/preview-generator/blob/4b058da08b893e044336b45923bfbded2f3a943f/preview_generator/preview/builder/image__imconvert.py#L152-L180

As i readded the mimetype/extension mapping, there should be no case where the guessing from mimetype fail (if there was no mapping, it's based on os data, which are know to be different between different computer).

Reply:
This bug has been fixed ：  946e66c176ad2feb337d23022872dc3f866e4948

Reply:
@inkhey  Thank you for your patience and help me find this problems. 

- The problem of recognition without extensions and the problem off xcf have been resolved
- A test case has also been fixed.

In addition, I think: For some special formats, such as xcf , add some test cases, such as judging whether the output image is close to the expected image.(Compare the two pictures directly). This may take a while, perhaps in another pr. You take a look.

Reply:
@a523 
Hello, thanks for all this work :+1: 
About the tests, i do create an issue: https://github.com/algoo/preview-generator/issues/286
For the heic issue, i tested it and the support is really system-related. On ubuntu 18.04, imagemagick doesn't handle this format whereas on debian bullseye it does work.

I have a last request before be able to merge it upstream, can you merge you code with develop in order to get last fix. This will permit to run the new (not public for now) ci on this branch too.

Reply:
That's all ok !